### PR TITLE
Docs/#1552 document controller snapshots

### LINKED ISSUE #1552: Document Controller Snapshots (23.2)
https://github.com/redpanda-data/core-internal/issues/132

Controller snapshots are somewhat of an implementation detail, but they'll show up for cluster admins and in support incidents as much faster cluster restart / start times, drastically reduce time to add a new node, etc. . This is good for admins to understand (as compared to the previous 'always load the entire controller log on startup' behavior of 23.1 and before)

Snapshots will eventually be used in our metadata backup/DR efforts , but this doc ticket is purely about highlighting how snapshotting works and how redpanda's controller log is managed in general (we can now load the snapshot and replay only incremental updates since the last snapshot to have new and restarted nodes become operational sooner

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1552 

This adds a new, small conceptual section called "Controller partition and snapshots" to the How Redpanda Works page. 

preview: https://deploy-preview-1646--redpanda-documentation.netlify.app/docs/beta/get-started/architecture/#controller-partition-and-snapshots

### CODE REVIEW

File: versioned_docs/version-23.2/get-started/architecture.mdx [hunk 1]
```diff
@@ -48,6 +48,12 @@ As long as there is a timing inequality between heartbeat time, election timeout
...
 
+## Controller partition and snapshots
+
+Redpanda stores metadata update commands (such as creating and deleting topics or users) in a system partition called the controller partition. Starting in version 23.2, controller snapshots save the current cluster metadata state to disk, so restoring from this snapshot only requires time proportional to the number of metadata objects. A new snapshot is created after each controller command is added.
```
Comment (Feediver1) [inline versioned_docs/version-23.2/get-started/architecture.mdx:53]: Given that this goes in the 23.2 docs, do you really need to state "Starting in version 23.2..." If we did this for every new feature, we would have A LOT of statements like this throughout our docs.

**Review by Feediver1 — APPROVED** [review]
Left comment--let's discuss in team meeting if you would like.

File: versioned_docs/version-23.2/get-started/architecture.mdx (hunk 1 above)
Comment (micheleRP) [reply versioned_docs/version-23.2/get-started/architecture.mdx:53]: I'll remove this!

**Comment by micheleRP** [comment]
Since Beta docs already published, I'm closing this PR and putting these changes in dev under `versioned_docs/version-23.2`.

**Pull request closed without merging**